src/algebra/lie/basic.lean

```diff
@@ -371,6 +453,54 @@ def inverse (f : M →ₗ⁅R,L⁆ N) (g : N → M)
               ... = ⁅x, g n⁆ : (h₁ _),
   ..linear_map.inverse f.to_linear_map g h₁ h₂ }
 
+instance : has_add (M →ₗ⁅R,L⁆ N) :=
+{ add := λ f g, { map_lie' := by simp, ..((f : M →ₗ[R] N) + (g : M →ₗ[R] N)) }, }
+
+instance : has_sub (M →ₗ⁅R,L⁆ N) :=
+{ sub := λ f g, { map_lie' := by simp, ..((f : M →ₗ[R] N) - (g : M →ₗ[R] N)) }, }
+
+instance : has_neg (M →ₗ⁅R,L⁆ N) :=
+{ neg := λ f, { map_lie' := by simp, ..(-(f : (M →ₗ[R] N))) }, }
+
+@[norm_cast, simp] lemma coe_add (f g : M →ₗ⁅R,L⁆ N) : ⇑(f + g) = f + g := rfl
+
+lemma add_apply (f g : M →ₗ⁅R,L⁆ N) (m : M) : (f + g) m = f m + g m := rfl
+
+@[norm_cast, simp] lemma coe_sub (f g : M →ₗ⁅R,L⁆ N) : ⇑(f - g) = f - g := rfl
+
+lemma sub_apply (f g : M →ₗ⁅R,L⁆ N) (m : M) : (f - g) m = f m - g m := rfl
+
+@[norm_cast, simp] lemma coe_neg (f : M →ₗ⁅R,L⁆ N) : ⇑(-f) = -f := rfl
+
+lemma neg_apply (f : M →ₗ⁅R,L⁆ N) (m : M) : (-f) m = -(f m) := rfl
+
+instance : add_comm_group (M →ₗ⁅R,L⁆ N) :=
+{ zero           := 0,
+  add            := (+),
+  neg            := has_neg.neg,
+  sub            := has_sub.sub,
+  nsmul          := λ n f, { map_lie' := λ x m, by simp, ..(n • (f : M →ₗ[R] N)) },
+  nsmul_zero'    := λ f, by { ext, simp, },
+  nsmul_succ'    := λ n f, by { ext, simp [nat.succ_eq_one_add, add_nsmul], },
+  ..(coe_injective.add_comm_group _ coe_zero coe_add coe_neg coe_sub :
+    add_comm_group (M →ₗ⁅R,L⁆ N)) }
+
+instance : has_scalar R (M →ₗ⁅R,L⁆ N) :=
+{ smul := λ t f, { map_lie' := by simp, ..(t • (f : M →ₗ[R] N)) }, }
+
+@[norm_cast, simp] lemma coe_smul (t : R) (f : M →ₗ⁅R,L⁆ N) : ⇑(t • f) = t • f := rfl
+
+lemma smul_apply (t : R) (f : M →ₗ⁅R,L⁆ N) (m : M) : (t • f) m = t • (f m) := rfl
+
+instance : module R (M →ₗ⁅R,L⁆ N) :=
+{ smul      := (•),
+  one_smul  := λ f, by { ext, simp only [smul_apply, one_smul], },
+  mul_smul  := λ t₁ t₂ f, by { ext, simp only [mul_smul, smul_apply], },
+  smul_add  := λ t f g, by { ext, simp only [smul_add, smul_apply, add_apply], },
+  smul_zero := λ t, by { ext, simp only [smul_apply, smul_zero, zero_apply], },
+  add_smul  := λ t₁ t₂ f, by { ext, simp only [smul_apply, add_apply, add_smul], },
+  zero_smul := λ f, by { ext, simp only [smul_apply, zero_smul, zero_apply], }, }
```

Comment:
```suggestion
function.injective.semimodule R ⟨to_fun, rfl, coe_add⟩ coe_injective coe_smul
```

Reply:
I was trying to get to this. You're too quick for me :-)